src/action.ts

```diff
@@ -0,0 +1,81 @@
+import { Link } from "./link";
+import { Device } from "./device";
+
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.*
+ */
+
+export interface IAction {
```

Comment:
I would rather name it `ActionDescription`.
The same applies to the other interfaces.